### PR TITLE
Add an option to split dprint output by risc

### PR DESCRIPTION
**Pull request by tt-dma**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/13081)

### Problem description
Multi-device/multi-core/multi-risc applications all have dprint data dumped to the same place, it's hard to parse.

### What's changed
Add an env var, `TT_METAL_DPRINT_ONE_FILE_PER_RISC`, to separate dprint output on a per-risc basis. Data is placed in generated files under `$TT_METAL_HOME/generated/dprint` as shown:
image

Also update dprint server to use logical coords instead of physical (and use CoreDescriptor to simplify the code) to facilitate this change.

### Checklist
- [ ] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/11042899735
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/impl/debug/dprint_server.cpp [hunk 1]
```diff
@@ -40,59 +41,93 @@ using namespace tt;
...
+#define CoreDescriptorSet set<CoreDescriptor, CoreDescriptorComparator>
+
 static inline float bfloat16_to_float(uint16_t bfloat_val) {
     uint32_t uint32_data = ((uint32_t)bfloat_val) << 16;
     float f;
     std::memcpy(&f, &uint32_data, sizeof(f));
     return f;
 }
 
+static std::string GetRiscName(CoreType core_type, int hart_id) {
```
Comment (pgkeller) [inline tt_metal/impl/debug/dprint_server.cpp:65]: coming soon (ish, I hope), this will all go into the HAL.  ok for now

**Review by pgkeller — APPROVED** [review]
Sometimes we want the prints separated, sometimes interlieved.  I've wondered for a while if we should preface prints to stdout w/ "BR: " or some such to help discern where the prints came from.  Then that begs the question: should core be printed as well, "(x,y,BR): ".

**Comment by tt-dma** [comment]
> Sometimes we want the prints separated, sometimes interlieved. I've wondered for a while if we should preface prints to stdout w/ "BR: " or some such to help discern where the prints came from. Then that begs the question: should core be printed as well, "(x,y,BR): ".

Makes sense, would say the tricky part here is knowing when to insert "(x,y,BR): ", since the dprint server handles "tokens" (a string, a number, a formatter, etc.). We'd need to expand the implementation to group dprints (by ENDL?). There is an issue filed for this: https://github.com/tenstorrent/tt-metal/issues/5945. Will leave a comment on there

**Comment by tt-dma** [comment]
New passing CI: https://github.com/tenstorrent/tt-metal/actions/runs/11079011337
